lib/class_kit/class_methods.rb

```diff
@@ -1,58 +1,55 @@
 module ClassKit
-  def attr_accessor_type(name, type: nil, collection_type: nil, allow_nil: true, default: nil, auto_init: false, meta: {})
+  def attr_accessor_type(name, type: nil, collection_type: nil, allow_nil: true, default: nil, auto_init: false,
+                         meta: {})
 
-    if !self.class_variable_defined?(:@@class_kit_attributes)
-      self.class_variable_set(:@@class_kit_attributes, [])
+    unless class_variable_defined?(:@@class_kit_attributes)
+      class_variable_set(:@@class_kit_attributes, [])
     end
 
-    if !self.class_variable_defined?(:@@is_class_kit_class)
-      self.class_variable_set(:@@is_class_kit_class, true)
-    end
+    attributes = class_variable_get(:@@class_kit_attributes)
 
-    attributes = self.class_variable_get(:@@class_kit_attributes)
-    attributes.push({ name: name, type: type, collection_type: collection_type, allow_nil: allow_nil, default: default, auto_init: auto_init, meta: meta })
-    self.class_variable_set(:@@class_kit_attributes, attributes)
+    attributes.push({ name: name, type: type, collection_type: collection_type, allow_nil: allow_nil,
+                                  default: default, auto_init: auto_init, meta: meta })
 
-    self.class_eval do
+    class_eval do
       define_method name do
-
         cka = self.class.class_variable_get(:@@class_kit_attributes).detect { |i| i[:name] == name }
```

Comment:
Using a Hash is a good idea and something I had considered, however if we did use a Hash we would need to potentially use 2 Hashes to store the data as the attribute helper currently allows attributes to be searched by `[:name]` and also by `[:type]` which means we need to find by more than 1 key. That said the find by `[:type]` is an edge case helper method and the majority of the code and use cases use the find by `[:name]` approach so maybe a compromise would be to keep the array but also maintain a Hash which uses the `[:name]` as the key for performance reasons that can be used for the majority of scenarios. I guess it comes down to what is worse the performance hit of using `.detect` on a small array (I would assume a class should not have more than 10-20 attributes at most if the single responsibility pattern is followed except in extreme cases) or duplicating the list of attributes into Hashes treated like indexes to allow retrieval by a specific key quickly (which would increase the memory footprint slightly and increase complexity).